components/clientComponents/forms/Review/reviewUtils.ts

```diff
@@ -0,0 +1,208 @@
+    filterValuesForShownElements(groupElements, shownFormElements)
+  );
+  return shownElementIds.map((elementId) => {
+    const element = formElements.find((item) => item.id === elementId);
+    const reviewElement: ReviewElement = {
+      title: element?.properties?.[getLocalizedProperty("title", lang)] as string,
+      values: formValues[elementId as unknown as keyof typeof formValues] as string,
+      element,
+    };
+
+    let resultValues: string | ReviewElement[] = formatElementValues(reviewElement);
+
+    // Handle any Sub Elements. Note Sub Elements = Dynamic Rows = Repeating Sets
+    if (element?.type === FormElementTypes.dynamicRow) {
+      resultValues = [];
+      const parentId = element.id;
+      const parentTitle = element.properties?.[getLocalizedProperty("title", lang)];
+      const subElements = element.properties?.subElements;
+      // Use FormValues as the source of truth and for each FormValue value, map the related
+      // subElement title to the FormValue value
+      const subElementValues = (formValues[parentId] as string[]).map(
+        (valueRows, valueRowsIndex) => {
+          const subElementsTitle = `${parentTitle} - ${valueRowsIndex + 1}`;
+          const valueRowsAsArray = Object.keys(valueRows).map(
+            (key) => valueRows[key as keyof typeof valueRows]
+          );
+          // Match the FormValue index to the subElement index to assign the Element title
+          const titlesMappedToValues = valueRowsAsArray.map((formValue, valueRowIndex) => {
+            return {
+              title: subElements?.[valueRowIndex].properties?.[getLocalizedProperty("title", lang)],
+              values: formValue,
+              element: element,
+            };
+          });
+          return {
+            title: subElementsTitle,
+            values: titlesMappedToValues,
+            element,
+          } as ReviewElement;
+        }
+      );
+      resultValues.push({
+        title: parentTitle as string,
+        values: subElementValues,
+        element,
+      });
+    }
+
+    // Handle Address Complete if broken into sub-components.
+    if (element?.type === FormElementTypes.addressComplete) {
+      const addressComponents = element.properties?.addressComponents as AddressComponents;
+
+      if (addressComponents && addressComponents.splitAddress === true) {
+        // We're a split address, broken into components.
+        const parentTitle = element.properties?.[getLocalizedProperty("title", lang)];
+
+        const addressFormValue = formValues[elementId] as string;
+        const addressValues = JSON.parse(addressFormValue) as AddressElements;
+        const canadaOnly = element.properties?.addressComponents?.canadianOnly;
+
+        const titleSet = {
+          streetAddress: parentTitle + " - " + addressCompleteStrings.streetAddress,
+          city: parentTitle + " - " + addressCompleteStrings.city,
+          province:
+            parentTitle +
```

Comment:
If we make these template string does it make it easier / less lines of code ?

Reply:
Yup sounds good. My hope is to move this code out and for it to be used as more of a customization. (future refactor) I never intended the code to be used this way.